### PR TITLE
Reply with json when not accepting html on LoginException

### PR DESCRIPTION
**Pull request by nickvergessen**
Follow up to #26572

### CODE REVIEW

**Review by korelstar — APPROVED** [review]
This fixes [the problem](https://github.com/nextcloud/server/pull/26572#issuecomment-830063651). Thanks for the quick response!

**Review by MorrisJobke — APPROVED** [review]
Makes sense 👍

File: index.php [hunk 1]
```diff
@@ -55,7 +55,19 @@
...
 } catch (\OC\User\LoginException $ex) {
-	OC_Template::printErrorPage($ex->getMessage(), $ex->getMessage(), 403);
+	$request = \OC::$server->getRequest();
+	/**
+	 * Routes with the @CORS annotation and other API endpoints should
+	 * not return a webpage, so we only print the error page when html is accepted,
+	 * otherwise we reply with a JSON array like the SecurityMiddleware would do.
+	 */
+	if (stripos($request->getHeader('Accept'),'html') === false) {
+		http_response_code(401);
```
Comment (MorrisJobke) [inline index.php:65]:
```
Scenario: Making a web request with a disabled user            # /drone/src/build/integration/features/provisioning-v1.feature:684
    Given As an "admin"                                          # FeatureContext::asAn()
    And user "user0" exists                                      # FeatureContext::assureUserExists()
    And assure user "user0" is disabled                          # FeatureContext::assureUserIsDisabled()
    And As an "user0"                                            # FeatureContext::asAn()
    When sending "GET" with exact url to "/index.php/apps/files" # FeatureContext::sendingToDirectUrl()
    And the HTTP status code should be "403"                     # FeatureContext::theHTTPStatusCodeShouldBe()
         Failed asserting that 401 matches expected '403'.
```

**Review by MorrisJobke — CHANGES_REQUESTED** [review]
See inline

File: index.php (hunk 1 above)
Comment (nickvergessen) [reply index.php:65]: Of course we have a test that makes sure we do send the wrong code...

Fixed

**Comment by korelstar** [comment]
Ping @MorrisJobke 
I'm looking forward for getting this merged :-)

**Review by LukasReschke — APPROVED** [review]

File: build/integration/features/provisioning-v1.feature [hunk 2]
```diff
@@ -237,8 +237,8 @@ Feature: provisioning
 		Given As an "brand-new-user"
 		When sending "POST" to "/cloud/users/brand-new-user/groups" with
 			| groupid | new-group |
-		Then the OCS status code should be "997"
-		And the HTTP status code should be "401"
+		Then the OCS status code should be "998"
```
Comment (MorrisJobke) [inline build/integration/features/provisioning-v1.feature:240]: CI says that this is still 997.

Comment (korelstar) [reply build/integration/features/provisioning-v1.feature:240]: I think 997 is correct, isn't it? Should I fix the test @nickvergessen ?

Comment (nickvergessen) [reply build/integration/features/provisioning-v1.feature]: arg, I pushed this to the wrong branch... I always wondered why https://github.com/nextcloud/server/pull/26679 never got green.

**Review by MorrisJobke — APPROVED** [review]
Fine with me 👍

**Pull request merged**